tests/sentry/utils/test_jwt.py

```diff
@@ -0,0 +1,189 @@
+    "kty": "RSA",
+}
+
+
+@pytest.fixture
+def token():
+    headers = {
+        "alg": "HS256",
+        "typ": "JWT",
+    }
+    claims = {
+        "iss": "me",
+    }
+    key = "secret"
+    encoded = pyjwt.encode(claims, key, algorithm="HS256", headers=headers)
+
+    # PyJWT < 2.0 returns bytes, not strings
+    token = encoded.decode("UTF-8")
+    assert isinstance(token, str)
+
+    return token
+
+
+def test_peek_claims(token):
+    claims = jwt_utils.peek_claims(token)
+    assert claims == {"iss": "me"}
+
+    for key, value in claims.items():
+        assert isinstance(key, str)
+        assert isinstance(value, str)
+
+
+def test_decode(token):
+    claims = jwt_utils.decode(token, "secret")
+    assert claims == {"iss": "me"}
+
+    for key, value in claims.items():
+        assert isinstance(key, str)
+        assert isinstance(value, str)
+
+    claims["aud"] = "you"
+    token = jwt_utils.encode(claims, "secret")
+
+    with pytest.raises(pyjwt.InvalidAudience):
+        jwt_utils.decode(token, "secret")
+
+
+def test_decode_audience():
+    payload = {
+        "iss": "me",
+        "aud": "you",
+    }
+    token = jwt_utils.encode(payload, "secret")
+
+    with pytest.raises(pyjwt.InvalidAudience):
+        jwt_utils.decode(token, "secret")
+
+    claims = jwt_utils.decode(token, "secret", audience="you")
+    assert claims == payload
+
+    with pytest.raises(pyjwt.InvalidAudience):
+        jwt_utils.decode(token, "secret", audience="wrong")
+
+    claims = jwt_utils.decode(token, "secret", audience=False)
+    assert claims == payload
+
+
+def test_encode(token):
+    headers = {
+        "alg": "HS256",
+        "typ": "JWT",
+    }
+    claims = {
+        "iss": "me",
+    }
+    key = b"secret"
+
+    encoded = jwt_utils.encode(claims, key, headers=headers)
+    assert isinstance(encoded, str)
+
+    assert encoded.count(".") == 2
+    assert encoded == token
+
+    decoded_claims = jwt_utils.decode(encoded, key)
+    assert decoded_claims == claims
+
+
+def test_encode_rs256():
+    headers = {
+        "alg": "RS256",
+        "typ": "JWT",
+    }
+    claims = {
+        "iss": "me",
+    }
+    encoded_hs256 = jwt_utils.encode(claims, "secret", headers=headers)
+    encoded_rs256 = jwt_utils.encode(claims, RS256_KEY, headers=headers, algorithm="RS256")
+
+    assert encoded_rs256.count(".") == 2
+    assert encoded_rs256 != encoded_hs256
+
+
+def test_authorization_header(token):
+    header = jwt_utils.authorization_header(token)
+    assert header == {"Authorization": f"Bearer {token}"}
+
+    header = jwt_utils.authorization_header(token, scheme="JWT")
+    assert header == {"Authorization": f"JWT {token}"}
+
+
+def test_rsa_key_from_jwk():
+    key = jwt_utils.rsa_key_from_jwk(json.dumps(RSA_JWK))
+    assert key
+    assert isinstance(key, RSAPrivateKey)
+
+    # Ensure we can use the key to create a token
+    claims = {"iss": "me"}
+    token = jwt_utils.encode(claims, key)
```

Comment:
This fails, I but I. think this is exactly what the existing code in `src/sentry/integrations/msteams/webhook.py` does currently so I don't really understand this (yes, the mypy types do not match but that doesn't affect runtime.  i was trying to get the runtime right before adding good types)

Reply:
The test passes if you wrap `key` in `str()`:
```
token = jwt_utils.encode(claims, str(key))
```

Reply:
Ah, this is what i was missing, thanks!

I've changed the tool to do the key into the bytes format so that it stays the same for `encode()` and `decode()` and the user doesn't have to deal with any conversions.  I think that makes more sense.